src/OpenTelemetry.Abstractions/Trace/Decision.cs

```diff
@@ -0,0 +1,62 @@
+﻿// <copyright file="Decision.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Trace
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Sampling decision.
+    /// </summary>
+    public struct Decision
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Decision"/> struct.
+        /// </summary>
+        /// <param name="isSampled">True if sampled, false otherwise.</param>
+        public Decision(bool isSampled)
+        {
+            this.IsSampled = isSampled;
+            this.Attributes = Enumerable.Empty<KeyValuePair<string, object>>();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Decision"/> struct.
+        /// </summary>
+        /// <param name="isSampled">True if sampled, false otherwise.</param>
+        /// <param name="attributes">Attributes associated with the sampling decision.</param>
+        public Decision(bool isSampled, IEnumerable<KeyValuePair<string, object>> attributes)
+        {
+            this.IsSampled = isSampled;
+
+            // TODO: it's a great thing to copy the list. But may not worth it as it
+            // makes another memory allocation.
+            this.Attributes = new List<KeyValuePair<string, object>>(attributes);
```

Comment:
I guess if user gives us a list and then modifies it - it's their problem and probably not legit scenario. So we don't want everyone's else perf to suffer to support this

I guess we can avoid copying it and do it later if will find a scenario where it's important and legit